consensus/types_internal.go

```diff
@@ -0,0 +1,197 @@
+package consensus
+
+import (
+	"github.com/deso-protocol/core/bls"
+	"github.com/deso-protocol/core/collections/bitset"
+	"github.com/holiman/uint256"
+)
+
+//////////////////////////////////////////////////////////
+// BlockHash interface implementation for testing
+//////////////////////////////////////////////////////////
+
+type blockHash struct {
```

Comment:
After reviewing, I still wasn't sure why we needed these "internal" type defs. Can you just add a comment at the top explaining why they're useful?

Reply:
They're just useful for unit testing and integration testing within the consensus module. The module doesn't pull in any types from lib, so it has to define its own barebones structs mostly for tests. I'll explain on the struct definition's comments to say this.